scripts/xtensa-build-zephyr.py

```diff
@@ -787,10 +787,14 @@ def build_platforms():
 		_dict = dataclasses.asdict(platform_configs[platform])
 		platform_dict = { k:v for (k,v) in _dict.items() if _dict[k] is not None }
 
+		xtensa_tools_version = os.getenv("XTENSA_TOOLS_VERSION")
+		if not xtensa_tools_version:
+			xtesna_tools_version = platform_dict["XTENSA_TOOLS_VERSION"]
```

Comment:
This is potentially a scary change. Whether hard coding the versions in this python file is good or bad, it's been in use for a long time and this has specifically been done to ensure all developers and the various build systems, use the same configuration to build. And more importantly, any change is done with a git commit that is tested with the same test process as other pull requests. If we now open a new getenv() path to modify the build, it's hard to ensure there are no CI builders out there, that have foobar XTENSA_TOOLS_VERSION contents and in worst-case, don't fail immediate but start generating slightly different binaries (and test runners test those).

So I'd like to make this more explicit. Use a different name for the environment variable, or at minimum print a prominent warning to log file that non-standard toolchain is used in the build. I don't think this is now printed at all.

Reply:
That's fair.  Will resubmit in isolation and we can resume discussion there.

Reply:
Resubmitted in #9736